### PR TITLE
`Development`: Rewrite e2e exercise logic to support multiple exercises of the same type

### PR DESCRIPTION
**Pull request by TheZoker**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

At the moment we support only one exercise type per exam. Only one text exercise, one modeling exercise etc.

### Description

This PR tries to refactor the underlying code of the E2E tests (and the E2E test itself) to allow for multiple exercise types per exam. A followup PR will add a test exam test, that tests a minimum 2 of each exercises for this test exam. 

#### Code Review
- [x] Review 1
- [ ] Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2
#### Exam Mode Test
- [ ] Test

### CODE REVIEW

File: src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-container.component.html [hunk 1]
```diff
@@ -12,126 +12,131 @@
...
->
-    <ng-container *ngIf="submission; then assessment; else noSubmission"></ng-container>
-</jhi-assessment-layout>
-
-<ng-template #assessment>
-    <ng-container
-        *jhiExtensionPoint="overrideCodeEditor; context: { title: editorTitle, toolbar: editorToolbar, feedback: editorBottom, exercise: exercise, participation: participation }"
+<div [id]="'exercise-' + exercise?.id">
+    <jhi-assessment-layout
+        (navigateBack)="navigateBack()"
```
Comment (simon-hng) [inline src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-container.component.html:17]: note: This element probably has to many `@inputs`

File: src/main/webapp/app/exercises/programming/participate/code-editor-student-container.component.html [hunk 2]
```diff
@@ -1,88 +1,91 @@
...
-        <span *ngIf="participationCouldNotBeFetched">
-            <fa-icon [icon]="faTimesCircle" class="text-danger"></fa-icon>
-            <span jhiTranslate="artemisApp.editor.errors.participationNotFound">Your participation could not be found.</span>
-        </span>
+<div [id]="'exercise-' + exercise?.id">
+    <nav *ngIf="loadingParticipation || participationCouldNotBeFetched" class="card">
+        <div class="card-body">
+            <span *ngIf="loadingParticipation">
+                <fa-icon [icon]="faCircleNotch" [spin]="true"></fa-icon>
+                <span jhiTranslate="artemisApp.editor.fetchingParticipation"> Your participation is being loaded... </span>
```
Comment (simon-hng) [inline src/main/webapp/app/exercises/programming/participate/code-editor-student-container.component.html:6]: IMO paragraph
```suggestion
                <p jhiTranslate="artemisApp.editor.fetchingParticipation"> Your participation is being loaded... </p>
```

File: src/main/webapp/app/exercises/programming/participate/code-editor-student-container.component.html [hunk 3]
```diff
@@ -1,88 +1,91 @@
...
+            </span>
+            <span *ngIf="participationCouldNotBeFetched">
+                <fa-icon [icon]="faTimesCircle" class="text-danger"></fa-icon>
+                <span jhiTranslate="artemisApp.editor.errors.participationNotFound">Your participation could not be found.</span>
```
Comment (simon-hng) [inline src/main/webapp/app/exercises/programming/participate/code-editor-student-container.component.html:10]:
```suggestion
                <p jhiTranslate="artemisApp.editor.errors.participationNotFound">Your participation could not be found.</p>
```

File: src/main/webapp/app/exercises/programming/participate/code-editor-student-container.component.html (hunk 2 above)
Comment (simon-hng) [reply src/main/webapp/app/exercises/programming/participate/code-editor-student-container.component.html:6]: although [docs](https://docs.artemis.cit.tum.de/dev/guidelines/client/#icons-with-text) suggests spans. I still think paragraphs are better for accessibility reasons

**Review by simon-hng — COMMENTED** [review]
Left some comments, otherwise looks good

File: src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-container.component.html (hunk 1 above)
Comment (TheZoker) [reply src/main/webapp/app/exercises/programming/assess/code-editor-tutor-assessment-container.component.html:17]: Although it does not really show in the git diff, the only thing I did was wrap the existing html code with one `div` that has the `id` on it. So changing the elements with the `div` would be something for a followup PR. But thanks for the hints!

File: src/main/webapp/app/exercises/programming/participate/code-editor-student-container.component.html (hunk 2 above)
Comment (TheZoker) [reply src/main/webapp/app/exercises/programming/participate/code-editor-student-container.component.html:6]: See comment above

File: src/main/webapp/app/exercises/programming/participate/code-editor-student-container.component.html (hunk 3 above)
Comment (TheZoker) [reply src/main/webapp/app/exercises/programming/participate/code-editor-student-container.component.html:10]: See comment above

**Review by simon-hng — APPROVED** [review]
Code

**Review by simon-hng — APPROVED** [review]

**Review by 4ludwig4 — APPROVED** [review]
Checked the E2E tests and looked at the code. Did a few samples if the introduced div tag has wrong indentations, but couldn't find any issues.

**Pull request merged**